workspaces/vault/plugins/vault-backend/config.d.ts

```diff
@@ -31,10 +31,42 @@ export interface Config {
     publicUrl?: string;
 
     /**
-     * The token used by Backstage to access Vault.
-     * @visibility secret
+     * The authentication used to connect to vault. Currently a static token and
+     * kubernetes authentication are supported.
+     * @visibility backend
```

Comment:
I would keep `token` to make the changes retro-compatible

```suggestion
    /**
     * The token used by Backstage to access Vault.
     * @visibility secret
     * @deprecated Use `auth` instead.
     */
     token: string;
    /**
     * The authentication used to connect to vault. Currently a static token and
     * kubernetes authentication are supported.
     * @visibility backend
```

Reply:
Thanks for the hint! I've just updated the code.

The `vault.token` can be used (and it has prio over the new config, just to keep it simple and not adding more if-else statements). However, a warning will be shown if this is used. Also added a tests to make sure the retro-compatibility is working.

I've kept it a `minor` change though, I think it makes sense as it's a new feature. I can change the release type to `patch` if you prefer

Reply:
yeah patch is good, as there is no api changes.
BTW I think some adjustments are needed in the config as right now both `token`  and `auth` are marked as required.

Reply:
Does it looks better now with the optional `vault.token`? I think this way we can enforce people to use the new `vault.auth` type, even though keeping their old setup won't make anything break for now